src/conic.jl

```diff
@@ -975,13 +981,16 @@ function MathProgBase.optimize!(m::PajaritoConicModel)
             if optimality_gap < (abs(mip_objval) + 1e-5)*m.acceptable_opt_tolerance
                 (m.verbose > 1) && println("MINLP Solved")
                 m.status = :Optimal
-                m.iterations = iter
                 (m.verbose > 1) && println("CPUTIME: $(time() - start)")
                 (m.verbose > 1) && println("Number of OA iterations: $iter")
+            else #cycle_indicator
+                m.status = :Suboptimal
             end
+            m.iterations = iter
             #break
         end
         (m.verbose > 0) && (m.algorithm == "OA") && @printf "%9d   %+.7e   %+.7e   %+.7e   %+.7e   %+.7e   %+.7e\n" iter mip_objval conic_objval optimality_gap m.objval primal_infeasibility OA_infeasibility
+        (cycle_indicator) && warn("Mixed-integer cycling detected, terminating Pajarito...")
```

Comment:
The condition should be ``(cycle_indicator && m.status != :Optimal)``. Right now it prints the warning on every solution!